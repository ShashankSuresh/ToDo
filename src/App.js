import React, { useState } from "react";
import ConstructList from "./Components/construct-list";
import AddItem from "./Components/add-to-list";
import CompleteList from "./Components/complete-list";
import "./App.scss";

function App() {
  const [todoList, createNewList] = useState([
    {
      text: "todo text 1",
      complete: true
    },
    {
      text: "todo text 2",
      complete: false
    },
    {
      text: "todo text 3",
      complete: false
    },
    {
      text: "todo text 4",
      complete: false
    },
    {
      text: "todo text 5",
      complete: false
    }
  ]);

  const clearInputFields = () => {
    document.getElementById("add").value = "";
  };

  const addNewItem = text => {
    const newItemObj = {
      text: text,
      complete: false
    };
    const updatedList = [...todoList, { ...newItemObj }];
    createNewList(updatedList);
    clearInputFields();
  };

  const removeItem = index => {
    const newItemObj = [...todoList];
    newItemObj.splice(index, 1);
    return createNewList(newItemObj);
  };

  const completeItem = index => {
    const newItemObj = [...todoList];
    todoList[index].complete = true;
    return createNewList(newItemObj);
  };

  const editItem = index => {
    const itemEl = document.getElementById("item-text-" + index);
    const editBtnEl = document.getElementById("edit-button-" + index);
    const saveBtnEl = document.getElementById("save-button-" + index);
    itemEl.setAttribute("contentEditable", true);
    itemEl.focus();
    editBtnEl.style.display = "none";
    saveBtnEl.style.display = "block";
  };

  const saveItem = index => {
    const newItemObj = [...todoList];
    const itemEl = document.getElementById("item-text-" + index);
    const editBtnEl = document.getElementById("edit-button-" + index);
    const itemElVal = itemEl.textContent;
    const saveBtnEl = document.getElementById("save-button-" + index);
    itemEl.setAttribute("contentEditable", false);
    editBtnEl.style.display = "block";
    saveBtnEl.style.display = "none";
    todoList[index].text = itemElVal;
    return createNewList(newItemObj);
  };

  return (
    <div className="todo">
      <h2>Add Item</h2>
      <AddItem addNewItemFn={addNewItem} />

      <h2>To Do</h2>
      <div className="container">
        {todoList.map((todo, index) => (
          <ConstructList
            key={index}
            index={index}
            listItem={todo}
            editItemFn={editItem}
            saveItemFn={saveItem}
            removeItemFn={removeItem}
            completeItemFn={completeItem}
          />
        ))}
      </div>

      <h2>Complete</h2>
      {todoList.map((todo, index) => (
        <CompleteList index={index} key={index} listItem={todo} />
      ))}
    </div>
  );
}

export default App;
