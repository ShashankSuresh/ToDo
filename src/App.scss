body {
    font-family: Arial, Helvetica, sans-serif;
    display: flex;
    justify-content: center;
    h2 {
        text-transform: uppercase;
        font-size: 16px;
        border-bottom: 2px solid black;
        padding-bottom: 15px;;
        margin-top: 40px;
    }
    .todo {
        .todo-container {
            margin-bottom: 20px;
            .text-elements-container {
                display: flex;
                .input-textarea {
                    font-size: 16px;
                    margin: 0;
                }
            }
            .btn-container {
                display: flex;
                margin: 10px 4px 4px 4px;
                .btn-edit, .btn-save, .btn-delete {
                   margin-right: 10px; 
                   padding: 6px;
                }
            }
        }
        .strike-through {
            text-decoration: line-through;
        }
        
        .hidden {
            display: none;
        }
        
        .not-hidden {
            display: block;
        }
    }

}