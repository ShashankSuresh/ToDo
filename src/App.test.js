import React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";

describe("Todo", () => {
  beforeEach(() => {
    const input = document.createElement("input");
    const itemEl = document.createElement("p");
    const editBtnEl = document.createElement("button");
    const saveBtnEl = document.createElement("button");

    input.setAttribute("id", "add");
    itemEl.setAttribute("id", "item-text-1");
    itemEl.setAttribute("contentEditable", true);
    editBtnEl.setAttribute("id", "edit-button-1");
    editBtnEl.setAttribute("display", "none");
    saveBtnEl.setAttribute("id", "save-button-1");
    saveBtnEl.setAttribute("display", "block");

    document.body.appendChild(itemEl);
    document.body.appendChild(editBtnEl);
    document.body.appendChild(saveBtnEl);
    document.body.appendChild(input);
  });

  it("renders", () => {
    shallow(<App />);
  });

  it("displays initial to-dos", () => {
    const wrapper = mount(<App />);
    expect(
      wrapper.find(".todo-container").find("div.todo-container")
    ).toHaveLength(4);
  });

  it("addNewItem", () => {
    const wrapper = mount(<App />);
    wrapper
      .find(".todo")
      .find("#add")
      .instance().value = "sample text";
    wrapper
      .find(".todo")
      .find("#add-button")
      .simulate("click");
    expect(
      wrapper
        .find(".todo")
        .find("#add")
        .instance().value
    ).toEqual("sample text");
    expect(
      wrapper.find(".todo-container").find("div.todo-container")
    ).toHaveLength(5);
  });

  it("removeItem", () => {
    const wrapper = mount(<App />);
    wrapper
      .find(".todo")
      .find(".todo-container")
      .at(0)
      .find(".btn-delete")
      .simulate("click");
    expect(
      wrapper.find(".todo-container").find("div.todo-container")
    ).toHaveLength(3);
  });

  it("completeItem", () => {
    const wrapper = mount(<App />);
    wrapper
      .find(".todo")
      .find(".todo-container")
      .at(0)
      .find(".checkbox-complete")
      .simulate("click");
    expect(
      wrapper.find(".todo-container").find("div.todo-container")
    ).toHaveLength(3);
  });

  it("editItem", () => {
    const wrapper = mount(<App />);
    wrapper
      .find(".todo")
      .find(".todo-container")
      .at(0)
      .find(".btn-edit")
      .simulate("click");
    expect(
      wrapper
        .find(".todo")
        .find(".todo-container")
        .at(0)
        .find("#item-text-1")
        .props()
    ).toEqual({
      suppressContentEditableWarning: "true",
      contentEditable: "false",
      id: "item-text-1",
      className: "input-textarea",
      children: "todo text 2"
    });
  });

  it("saveItem", () => {
    const wrapper = mount(<App />);
    wrapper
      .find(".todo")
      .find(".todo-container")
      .at(0)
      .find(".btn-save")
      .simulate("click");
    expect(
      wrapper
        .find(".todo")
        .find(".todo-container")
        .at(0)
        .find("#item-text-1")
        .props()
    ).toEqual({
      suppressContentEditableWarning: "true",
      contentEditable: "false",
      id: "item-text-1",
      className: "input-textarea",
      children: ""
    });
  });
});
