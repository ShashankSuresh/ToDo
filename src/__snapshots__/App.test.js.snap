// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App with props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo"
      >
        <h2>
          Add Item
        </h2>
        <div>
          <input
            id="add"
            type="text"
          />
          <button>
            Add
          </button>
        </div>
        <h2>
          To Do
        </h2>
        <ul>
          <div
            class="class-name"
            id="todo-item-1"
          >
            <input
              type="checkbox"
            />
            <p
              contenteditable="false"
              id="item-text-1"
            >
              sample 2
            </p>
            <div>
              <button
                class="not-hidden"
                id="edit-button-1"
              >
                Edit
              </button>
              <button
                class="hidden"
                id="save-button-1"
              >
                Save
              </button>
              <button>
                Delete
              </button>
            </div>
          </div>
          <div
            class="class-name"
            id="todo-item-2"
          >
            <input
              type="checkbox"
            />
            <p
              contenteditable="false"
              id="item-text-2"
            >
              sample 3
            </p>
            <div>
              <button
                class="not-hidden"
                id="edit-button-2"
              >
                Edit
              </button>
              <button
                class="hidden"
                id="save-button-2"
              >
                Save
              </button>
              <button>
                Delete
              </button>
            </div>
          </div>
          <div
            class="class-name"
            id="todo-item-3"
          >
            <input
              type="checkbox"
            />
            <p
              contenteditable="false"
              id="item-text-3"
            >
              sample 4
            </p>
            <div>
              <button
                class="not-hidden"
                id="edit-button-3"
              >
                Edit
              </button>
              <button
                class="hidden"
                id="save-button-3"
              >
                Save
              </button>
              <button>
                Delete
              </button>
            </div>
          </div>
          <div
            class="class-name"
            id="todo-item-4"
          >
            <input
              type="checkbox"
            />
            <p
              contenteditable="false"
              id="item-text-4"
            >
              sample 5
            </p>
            <div>
              <button
                class="not-hidden"
                id="edit-button-4"
              >
                Edit
              </button>
              <button
                class="hidden"
                id="save-button-4"
              >
                Save
              </button>
              <button>
                Delete
              </button>
            </div>
          </div>
        </ul>
        <h2>
          Complete
        </h2>
        <ul>
          <div>
            <input
              checked=""
              disabled=""
              type="checkbox"
            />
            <span
              class="strike-through"
            >
              sample 1
            </span>
          </div>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo"
    >
      <h2>
        Add Item
      </h2>
      <div>
        <input
          id="add"
          type="text"
        />
        <button>
          Add
        </button>
      </div>
      <h2>
        To Do
      </h2>
      <ul>
        <div
          class="class-name"
          id="todo-item-1"
        >
          <input
            type="checkbox"
          />
          <p
            contenteditable="false"
            id="item-text-1"
          >
            sample 2
          </p>
          <div>
            <button
              class="not-hidden"
              id="edit-button-1"
            >
              Edit
            </button>
            <button
              class="hidden"
              id="save-button-1"
            >
              Save
            </button>
            <button>
              Delete
            </button>
          </div>
        </div>
        <div
          class="class-name"
          id="todo-item-2"
        >
          <input
            type="checkbox"
          />
          <p
            contenteditable="false"
            id="item-text-2"
          >
            sample 3
          </p>
          <div>
            <button
              class="not-hidden"
              id="edit-button-2"
            >
              Edit
            </button>
            <button
              class="hidden"
              id="save-button-2"
            >
              Save
            </button>
            <button>
              Delete
            </button>
          </div>
        </div>
        <div
          class="class-name"
          id="todo-item-3"
        >
          <input
            type="checkbox"
          />
          <p
            contenteditable="false"
            id="item-text-3"
          >
            sample 4
          </p>
          <div>
            <button
              class="not-hidden"
              id="edit-button-3"
            >
              Edit
            </button>
            <button
              class="hidden"
              id="save-button-3"
            >
              Save
            </button>
            <button>
              Delete
            </button>
          </div>
        </div>
        <div
          class="class-name"
          id="todo-item-4"
        >
          <input
            type="checkbox"
          />
          <p
            contenteditable="false"
            id="item-text-4"
          >
            sample 5
          </p>
          <div>
            <button
              class="not-hidden"
              id="edit-button-4"
            >
              Edit
            </button>
            <button
              class="hidden"
              id="save-button-4"
            >
              Save
            </button>
            <button>
              Delete
            </button>
          </div>
        </div>
      </ul>
      <h2>
        Complete
      </h2>
      <ul>
        <div>
          <input
            checked=""
            disabled=""
            type="checkbox"
          />
          <span
            class="strike-through"
          >
            sample 1
          </span>
        </div>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo"
      >
        <h2>
          Add Item
        </h2>
        <div>
          <input
            id="add"
            type="text"
          />
          <button>
            Add
          </button>
        </div>
        <h2>
          To Do
        </h2>
        <ul>
          <div
            class="class-name"
            id="todo-item-1"
          >
            <input
              type="checkbox"
            />
            <p
              contenteditable="false"
              id="item-text-1"
            >
              sample 2
            </p>
            <div>
              <button
                class="not-hidden"
                id="edit-button-1"
              >
                Edit
              </button>
              <button
                class="hidden"
                id="save-button-1"
              >
                Save
              </button>
              <button>
                Delete
              </button>
            </div>
          </div>
          <div
            class="class-name"
            id="todo-item-2"
          >
            <input
              type="checkbox"
            />
            <p
              contenteditable="false"
              id="item-text-2"
            >
              sample 3
            </p>
            <div>
              <button
                class="not-hidden"
                id="edit-button-2"
              >
                Edit
              </button>
              <button
                class="hidden"
                id="save-button-2"
              >
                Save
              </button>
              <button>
                Delete
              </button>
            </div>
          </div>
          <div
            class="class-name"
            id="todo-item-3"
          >
            <input
              type="checkbox"
            />
            <p
              contenteditable="false"
              id="item-text-3"
            >
              sample 4
            </p>
            <div>
              <button
                class="not-hidden"
                id="edit-button-3"
              >
                Edit
              </button>
              <button
                class="hidden"
                id="save-button-3"
              >
                Save
              </button>
              <button>
                Delete
              </button>
            </div>
          </div>
          <div
            class="class-name"
            id="todo-item-4"
          >
            <input
              type="checkbox"
            />
            <p
              contenteditable="false"
              id="item-text-4"
            >
              sample 5
            </p>
            <div>
              <button
                class="not-hidden"
                id="edit-button-4"
              >
                Edit
              </button>
              <button
                class="hidden"
                id="save-button-4"
              >
                Save
              </button>
              <button>
                Delete
              </button>
            </div>
          </div>
        </ul>
        <h2>
          Complete
        </h2>
        <ul>
          <div>
            <input
              checked=""
              disabled=""
              type="checkbox"
            />
            <span
              class="strike-through"
            >
              sample 1
            </span>
          </div>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo"
    >
      <h2>
        Add Item
      </h2>
      <div>
        <input
          id="add"
          type="text"
        />
        <button>
          Add
        </button>
      </div>
      <h2>
        To Do
      </h2>
      <ul>
        <div
          class="class-name"
          id="todo-item-1"
        >
          <input
            type="checkbox"
          />
          <p
            contenteditable="false"
            id="item-text-1"
          >
            sample 2
          </p>
          <div>
            <button
              class="not-hidden"
              id="edit-button-1"
            >
              Edit
            </button>
            <button
              class="hidden"
              id="save-button-1"
            >
              Save
            </button>
            <button>
              Delete
            </button>
          </div>
        </div>
        <div
          class="class-name"
          id="todo-item-2"
        >
          <input
            type="checkbox"
          />
          <p
            contenteditable="false"
            id="item-text-2"
          >
            sample 3
          </p>
          <div>
            <button
              class="not-hidden"
              id="edit-button-2"
            >
              Edit
            </button>
            <button
              class="hidden"
              id="save-button-2"
            >
              Save
            </button>
            <button>
              Delete
            </button>
          </div>
        </div>
        <div
          class="class-name"
          id="todo-item-3"
        >
          <input
            type="checkbox"
          />
          <p
            contenteditable="false"
            id="item-text-3"
          >
            sample 4
          </p>
          <div>
            <button
              class="not-hidden"
              id="edit-button-3"
            >
              Edit
            </button>
            <button
              class="hidden"
              id="save-button-3"
            >
              Save
            </button>
            <button>
              Delete
            </button>
          </div>
        </div>
        <div
          class="class-name"
          id="todo-item-4"
        >
          <input
            type="checkbox"
          />
          <p
            contenteditable="false"
            id="item-text-4"
          >
            sample 5
          </p>
          <div>
            <button
              class="not-hidden"
              id="edit-button-4"
            >
              Edit
            </button>
            <button
              class="hidden"
              id="save-button-4"
            >
              Save
            </button>
            <button>
              Delete
            </button>
          </div>
        </div>
      </ul>
      <h2>
        Complete
      </h2>
      <ul>
        <div>
          <input
            checked=""
            disabled=""
            type="checkbox"
          />
          <span
            class="strike-through"
          >
            sample 1
          </span>
        </div>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
